packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/flutter_provider_interfaces/device_context_info_provider.dart

```diff
@@ -0,0 +1,187 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:amplify_core/amplify_core.dart';
+
+/// {@template amplify_analytics_pinpoint_dart.device_context_info_provider}
+/// Provide information required for Pinpoint EndpointDemographic object
+///
+/// For more details see [Pinpoint Endpoint](https://docs.aws.amazon.com/pinpoint/latest/apireference/apps-application-id-endpoints.html)online spec.
+/// {@endtemplate}
+abstract class DeviceContextInfoProvider {
+  /// {@macro amplify_analytics_pinpoint_dart.device_context_info_provider}
+  Future<DeviceContextInfo> getDeviceInfoDetails();
+}
+
+/// {@template amplify_analytics_pinpoint_dart.device_context_info.sent_to_pinpoint_endpoint}
+/// Sent as a field to [Pinpoint Endpoint](https://docs.aws.amazon.com/pinpoint/latest/apireference/apps-application-id-endpoints.html).
+/// {@endtemplate}
+
+/// {@template amplify_analytics_pinpoint_dart.device_context_info.sent_to_pinpoint_event}
+/// Sent as a field to [Pinpoint Event](https://docs.aws.amazon.com/pinpoint/latest/apireference/apps-application-id-events.html).
+/// {@endtemplate}
+
+/// {@template amplify_analytics_pinpoint_dart.device_context_info}
+/// Data representation of all device information sent to Pinpoint EndpointDemographic
+///
+/// For more details see [Pinpoint Endpoint](https://docs.aws.amazon.com/pinpoint/latest/apireference/apps-application-id-endpoints.html)online spec.
+/// {@endtemplate}
```

Comment:
I'm not sure what the intention here is, but all of these comments are included in the doc comment for `DeviceContextInfo`. The first three templates all become one paragraph because there is no white space. Is that what you want?
<img width="791" alt="Screen Shot 2022-10-13 at 3 15 24 PM" src="https://user-images.githubusercontent.com/20613561/195689522-15000961-19f9-444b-85a1-4c6bd57b6947.png">

Reply:
No thanks for catching that.  Moved declaration to first time use